staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -1202,7 +1203,13 @@ func azToRegion(az string) (string, error) {
 	if len(az) < 1 {
 		return "", fmt.Errorf("invalid (empty) AZ")
 	}
-	region := az[:len(az)-1]
+
+	r := regexp.MustCompile(`\w+-\w+-\d+`)
```

Comment:
there are some with 3 words in the code like x-x-x-1

Reply:
submitted too early I meant like us-gov-east-1  https://aws.amazon.com/govcloud-us/?whats-new-ess.sort-by=item.additionalFields.postDateTime&whats-new-ess.sort-order=desc https://docs.aws.amazon.com/govcloud-us/latest/UserGuide/using-govcloud-endpoints.html

Reply:
sorry submitted too early I meant like us-gov-east-1 (govcloud) https://docs.aws.amazon.com/govcloud-us/latest/UserGuide/using-govcloud-endpoints.html

Reply:
ah, I will catch this case. So there's all the cases we have? Any other edge cases? 

- us-gov-east-1
- us-west-2-lax-1a
- us-east-1

Let me double check codes and see if we can get rid of `azToRegion` entirely and fetch region from ec2metadata service directly. Then we don't have to deal with these cases

Reply:
ec2metadata might not be available from masters :D (we'll block access to it)
The current possible region regexs is https://github.com/boto/botocore/blob/master/botocore/data/endpoints.json (different partition have different ones) :D

Reply:
@M00nF1sh This is helpful. :D